src/ai/backend/storage/kmanila/__init__.py

```diff
@@ -0,0 +1,272 @@
+        kmanila_requestor_pwd: str,
+        default_project_id: str | None,
+        default_netword_id: str | None,
+        max_poll_count: int,
+    ) -> None:
+        super().__init__(mount_path)
+        self.etcd = etcd
+        self.api_base_url = api_base_url
+        self.agent_addrs = agent_addrs
+        self.access_to = access_to
+        self.access_level = access_level
+        self.kmanila_requestor_id = kmanila_requestor_id
+        self.kmanila_requestor_pwd = kmanila_requestor_pwd
+        self.default_project_id = default_project_id
+        self.default_netword_id = default_netword_id
+        self.max_poll_count = max_poll_count
+
+    async def _get_auth_info(self, user_id: QuotaScopeID) -> dict[str, Any]:
+        async with aiohttp.ClientSession() as sess:
+            rqst_id = self.kmanila_requestor_id
+            rqst_pwd = self.kmanila_requestor_pwd
+            request_body = {
+                "auth": {
+                    "identity": {
+                        "methods": ["password"],
+                        "password": {
+                            "user": {
+                                "domain": {"id": "default"},
+                                "name": rqst_id,
+                                "password": rqst_pwd,
+                            }
+                        },
+                    },
+                    "scope": {
+                        "project": {
+                            "domain": {"id": "default"},
+                            "name": rqst_id,
+                        }
+                    },
+                }
+            }
+            headers = {"Content-Type": "application/json", "Accept": "application/json"}
+            async with sess.post(
+                self.api_base_url / "d3/identity/auth/tokens", headers=headers, json=request_body
+            ) as resp:
+                token = resp.headers.get("X-Subject-Token")
+                if token is None:
+                    status_code = resp.status
+                    raise ExecutionError(f"Token not found. {status_code = }, {rqst_id = }")
+                if self.default_project_id is None:
+                    data: dict[str, Any] = await resp.json()
+                    project_id = data["token"]["project"]["id"]
+                else:
+                    project_id = self.default_project_id
+                return {"project_id": project_id, "auth_token": token}
+
+    async def get_user_volume_id(self, quota_scope_id: QuotaScopeID) -> str | None:
+        user_vol = await self.etcd.get_prefix(
+            f"{QUOTA_VOLUME_ID_MAP_KEY}/{quota_scope_id.scope_id}"
+        )
+        if not user_vol:
+            return None
+        return str(user_vol["volume_id"])
+
+    async def put_user_volume_id(self, quota_scope_id: QuotaScopeID, data: QuotaVolumeMap) -> None:
+        await self.etcd.put_prefix(f"{QUOTA_VOLUME_ID_MAP_KEY}/{quota_scope_id.scope_id}", data)  # type: ignore[arg-type]
```

Comment:
`type: ignore` annotation should be avoided in best effort. If there is a critical reason to bypass the type checker please describe the reason at the line.